components/script/dom/blob.rs

```diff
@@ -223,6 +230,66 @@ impl BlobMethods for Blob {
         let blob_impl = BlobImpl::new_sliced(rel_pos, self.blob_id.clone(), type_string);
         Blob::new(&*self.global(), blob_impl)
     }
+
+    // https://w3c.github.io/FileAPI/#text-method-algo
+    fn Text(&self) -> Rc<Promise> {
+        let global = self.global();
+        let in_realm_proof = AlreadyInRealm::assert(&global);
+        let p = Promise::new_in_current_realm(&global, InRealm::Already(&in_realm_proof));
+        let id = self.get_blob_url_id();
+        global.read_file_async(
+            id,
+            p.clone(),
+            Box::new(|promise, bytes| match bytes {
+                Ok(b) => {
+                    let (text, _, _) = UTF_8.decode(&b);
+                    let text = DOMString::from(text);
+                    promise.resolve_native(&text);
+                },
+                Err(e) => {
+                    promise.reject_error(e);
+                },
+            }),
+        );
+        p
+    }
+
+    // https://w3c.github.io/FileAPI/#arraybuffer-method-algo
+    #[allow(unsafe_code)]
+    fn ArrayBuffer(&self) -> Rc<Promise> {
+        let global = self.global();
+        let in_realm_proof = AlreadyInRealm::assert(&global);
+        let p = Promise::new_in_current_realm(&global, InRealm::Already(&in_realm_proof));
+
+        let id = self.get_blob_url_id();
+
+        global.read_file_async(
+            id,
+            p.clone(),
+            Box::new(|promise, bytes| {
+                match bytes {
+                    Ok(b) => {
+                        let cx = promise.global().get_cx();
+                        unsafe {
```

Comment:
Here I think we can reuse [`run_array_buffer_data_algorithm`](https://github.com/servo/servo/blob/12693b51f510761d5dd58a9103596a192edb803c/components/script/body.rs#L128)

For both suggestions, to see how the results can be resolved into a promise can be seen at https://github.com/servo/servo/blob/12693b51f510761d5dd58a9103596a192edb803c/components/script/body.rs#L95

Reply:
Also note that in the case of an error being returned by that function, I think it would be more appropriate to reject the promise with it, versus panicking.